yt_dlp/extractor/drtv.py

```diff
@@ -170,15 +171,25 @@ def _real_extract(self, url):
             programcard_url = '%s/%s' % (_PROGRAMCARD_BASE, video_id)
         else:
             programcard_url = _PROGRAMCARD_BASE
-            page = self._parse_json(
-                self._search_regex(
-                    r'data\s*=\s*({.+?})\s*(?:;|</script)', webpage,
-                    'data'), '1')['cache']['page']
-            page = page[list(page.keys())[0]]
-            item = try_get(
-                page, (lambda x: x['item'], lambda x: x['entries'][0]['item']),
-                dict)
-            video_id = item['customId'].split(':')[-1]
+            is_radio_url = 'dr.dk/lyd' in url
+            if is_radio_url:
+                json_string = self._search_regex(
+                    r'__NEXT_DATA__[^>]*>(.*)<\/script>',
+                    webpage,
+                    'next_data'
+                )
+                json_blob = self._parse_json(json_string, '1')
+                video_id = json_blob['props']['pageProps']['episode']['productionNumber']
+            else:
+                page = self._parse_json(
+                    self._search_regex(
+                        r'data\s*=\s*({.+?})\s*(?:;|</script)', webpage,
+                        'data'), '1')['cache']['page']
+                page = page[list(page.keys())[0]]
+                item = try_get(
+                    page, (lambda x: x['item'], lambda x: x['entries'][0]['item']),
+                    dict)
+                video_id = item['customId'].split(':')[-1]
```

Comment:
I know this is just old code / not yours, but it could be modernized/simplified with `_search_json` and `traverse_obj`
```suggestion
            else:
                json_data = self._search_json(
                    r'data\s*=', webpage, 'data', raw_video_id)
                video_id = traverse_obj(json_data, (
                    'cache', 'page', ..., (None, ('entries', 0)), 'item', 'customId',
                    {lambda x: x.split(':')[-1]}), get_all=False)
                if not video_id:
                    raise ExtractorError('Unable to extract video id')
```